internal/controllers/addon/controller.go

```diff
@@ -229,7 +229,15 @@ func (r *AddonReconciler) Reconcile(
 		return r.handleExit(requeueResult), nil
 	}
 
-	// Phase 5.
+	// Phase 5
+	// Ensure additional catalog sources
+	if requeueResult, err := r.ensureAdditionalCatalogSources(ctx, log, addon); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to ensure additional catalog source: %w", err)
+	} else if requeueResult != resultNil {
+		return r.handleExit(requeueResult), nil
+	}
+
```

Comment:
I see that the additional catalog sources are reconciled before the main catalog source. Does the ordering matter?

Might want to be consistent with the existing order in mt-tooling (first main, then additional) - https://github.com/mt-sre/managed-tenants-cli/blob/main/managedtenants/data/selectorsyncset.yaml.j2#L101-#L139

Reply:
For the odf use case, the additional catalog source/s bring in the dependencies for the main addon, hence I've decided to create them before the main catalog source

Reply:
Sure, made the change.